backend/core/src/main/java/org/sonarsource/sonarlint/core/plugin/PluginsService.java

```diff
@@ -164,23 +165,34 @@ private Set<Path> getPluginPathsForConnection(String connectionId) {
   }
 
   private Map<String, Path> getEmbeddedPluginPathsByKey(String connectionId) {
+    var embeddedPlugins = new HashMap<>(connectedModeEmbeddedPluginPathsByKey);
     if (supportsCustomSecrets(connectionId)) {
-      var embeddedPluginsExceptSecrets = new HashMap<>(connectedModeEmbeddedPluginPathsByKey);
-      embeddedPluginsExceptSecrets.remove(SonarLanguage.SECRETS.getPluginKey());
-      return embeddedPluginsExceptSecrets;
+      embeddedPlugins.remove(SonarLanguage.SECRETS.getPluginKey());
     }
-    return connectedModeEmbeddedPluginPathsByKey;
+    if (supportsIaCEnterprise(connectionId)) {
+      // if iacenterprise is there on the server, download both, iac and iacenterprise
+      embeddedPlugins.remove(SonarLanguage.AZURERESOURCEMANAGER.getPluginKey());
+    }
+    return embeddedPlugins;
+  }
+
+  public boolean supportsIaCEnterprise(String connectionId) {
+    return isSonarQubeCloudOrVersionOlderThan(ENTERPRISE_IAC_MIN_SQ_VERSION, connectionId);
   }
 
   public boolean supportsCustomSecrets(String connectionId) {
+    return isSonarQubeCloudOrVersionOlderThan(CUSTOM_SECRETS_MIN_SQ_VERSION, connectionId);
+  }
+
+  private boolean isSonarQubeCloudOrVersionOlderThan(Version version, String connectionId) {
```

Comment:
```suggestion
  private boolean isSonarQubeCloudOrVersionHigherThan(Version version, String connectionId) {
```
If I'm not mistaken, the check on the SQS version means that it should be higher, not older than the one provided as an argument 😉